crates/biome_cli/src/configuration.rs

```diff
@@ -120,6 +120,13 @@ impl LoadedConfiguration {
             }
         }
 
+        if let Some(_) = self.configuration.formatter.as_ref().and_then(|f| f.indent_size) {
+            let diagnostic = DeprecatedArgument { message: "The option indent-size is deprecated. Use indent-width instead.".to_string() };
+            console.error(markup! {
+                {if verbose { PrintDiagnostic::verbose(&diagnostic) } else { PrintDiagnostic::simple(&diagnostic) }}
+            })
+        }
+
```

Comment:
Hey @Levieber , I see now the issue :) 

So, my suggestion was actually wrong, and I can see why now. The PR really helped. The reason why you always see the diagnostic is because in this part of the code, `indent_size` was already filled with the **default value**, if the user didn't provide one.


I believe the deprecation diagnostic should be emitted **when we parse the configuration** file.

Here is the place:

https://github.com/biomejs/biome/blob/f468911604e2d11d8f1ed73e4f2ce2db8d1d1673/crates/rome_service/src/configuration/parse/json/formatter.rs#L45-L47

Still, you need to keep `DeprecatedArgument` because that will used **when the value is provided via CLI**.

Reply:
I guess I don't understand what I have to do.

Reply:
I would start by creating a new property called `indent_with`, don't worry about the diagnostics for now. Here's a similar PR where we add a new property: https://github.com/rome/tools/pull/4667

1. Start by adding a new property in the configuration

https://github.com/biomejs/biome/blob/8ae75955b357d4c6404fbb1e648a4014a5cfd4d6/crates/rome_service/src/configuration/formatter.rs#L30-L32

2. Then add the parsing logic, like here:
https://github.com/biomejs/biome/blob/8ae75955b357d4c6404fbb1e648a4014a5cfd4d6/crates/rome_service/src/configuration/parse/json/formatter.rs#L45-L47

```rs
 "indentWidth" => { 
     self.indent_width = self.map_to_u8(&value, name_text, u8::MAX, diagnostics); 
 }
```

3. Then, update the merging logic. This happens in two places. 

First place. This is where we merge the CLI arguments with the configuration coming from file. CLI has precedence.

https://github.com/biomejs/biome/blob/8ae75955b357d4c6404fbb1e648a4014a5cfd4d6/crates/rome_service/src/configuration/formatter.rs#L81-L83

Second place. This is where we take the configuration from different sources (defaults, config file, CLI) and pass the final value to the Workspace (our central control). Here, we want to **first** read the new value `indent_with`, then fallback to `indent_size`

https://github.com/biomejs/biome/blob/8ae75955b357d4c6404fbb1e648a4014a5cfd4d6/crates/rome_service/src/configuration/formatter.rs#L110

Start with these steps, then we can take it from there.